File: rocket-bi-web/di-core/data-ingestion/domain/response/JdbcSource.ts
/*
 * @author: tvc12 - Thien Vi
 * @created: 6/1/21, 2:17 PM
 */

import { DataSourceType, DataSources } from '@core/data-ingestion';
import { DataSource } from '@core/data-ingestion/domain/response/DataSource';
import { SourceId } from '@core/common/domain';

export class JdbcSource implements DataSource {
  id: SourceId;
  orgId: string;
  databaseType: DataSourceType;
  displayName: string;
  jdbcUrl: string;
  username: string;
  password: string;
  lastModify: number;
  readonly className = DataSources.JdbcSource;

  constructor(
    id: SourceId,
    orgId: string,
    databaseType: DataSourceType,
    displayName: string,
    jdbcURL: string,
    username: string,
    password: string,
    lastModify: number
  ) {
    this.id = id;
    this.orgId = orgId;
    this.displayName = displayName;
    this.databaseType = databaseType;
    this.jdbcUrl = jdbcURL;
    this.username = username;
    this.password = password;
    this.lastModify = lastModify;
  }

  static fromObject(obj: any): JdbcSource {
    return new JdbcSource(obj.id, obj.orgId, obj.databaseType, obj.displayName, obj.jdbcUrl, obj.username, obj.password, obj.lastModify);
  }
}

File: rocket-bi-server/src/main/scala/co/datainsider/jobscheduler/domain/source/JdbcSource.scala
package co.datainsider.jobscheduler.domain.source

import co.datainsider.jobscheduler.domain.DatabaseType.DatabaseType
import co.datainsider.jobscheduler.domain.DatabaseTypeRef
import co.datainsider.jobscheduler.domain.Ids.SourceId
import co.datainsider.jobscheduler.domain.source.DataSourceType.DataSourceType
import com.fasterxml.jackson.module.scala.JsonScalaEnumeration
import com.twitter.finatra.validation.constraints.NotEmpty

case class JdbcSource(
    orgId: Long,
    id: Long = 0,
    @NotEmpty displayName: String,
    @JsonScalaEnumeration(classOf[DatabaseTypeRef]) databaseType: DatabaseType,
    @NotEmpty jdbcUrl: String,
    @NotEmpty username: String,
    password: String,
    creatorId: String = "",
    lastModify: Long = System.currentTimeMillis()
) extends DataSource {

  override def getId: SourceId = id

  override def getName: String = displayName

  override def getType: DataSourceType = DataSourceType.Jdbc

  /** *
    *
    * @return config of this source to be pass to JobWorker, data is in form of key value pair, key values are hardcoded
    */
  override def getConfig: Map[String, Any] = {
    Map("database_type" -> databaseType.toString, "jdbc_url" -> jdbcUrl, "username" -> username, "password" -> password)
  }

  override def getCreatorId: String = creatorId

  override def getLastModify: SourceId = lastModify
}
